third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -1467,12 +1462,16 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                                 .iter()
                                 .map(|pat| self.get_node_type(pat.node_id()))
                                 .collect_vec();
-                            self.freeze_tuple_exp(&lhs_tys, rhs_tys, &mut rhs, &loc);
+                            self.freeze_tuple_exp(&lhs_tys, rhs_tys, rhs, &loc)
+                        } else {
+                            self.try_freeze(&lhs_ty, &rhs_ty, rhs)
                         }
+                    } else {
```

Comment:
Don't waste time here, but in the future, you can get rid of some of these redundant `else` clauses by combining matches in a tuple, e.g.:
```
if let (Pattern::Tuple(_, lhs_tys), Type::Tuple(rhs_tys)) = (&lhs, &rhs_ty) {
    ...
}
```
With dependencies it doesn't work as well (e.g., the innermost `lhs_tys`.) and sometimes it can move things too far apart, so don't go overboard.

Btw, it would be really good if you do other than just repeat the same variable name.  Why is one `lhs_tys` a `&Vec<Pattern>` and another one a `Vec<Type>`?  The first one probably should be called `lhs_pats` instead.